lm_eval/api/task.py

```diff
@@ -671,9 +671,9 @@ def __init__(
             check_choices = [test_target]
 
         for choice in check_choices:
-            choice_has_whitespace = True if " " in choice else False
+            choice_has_whitespace = True if choice.startswith(" ") or choice.endswith(" ") else False
```

Comment:
Maybe follow what was suggested by @baberabb  here: https://github.com/EleutherAI/lm-evaluation-harness/pull/798#discussion_r1301735313 ? Since we could also have trailing newlines, not just spaces, and want to flag that

Reply:
Yeah that's a good idea. I'll add that.

Reply:
Or actually, I'll just revert this and let the issue be solved in #798

Reply:
sgtm!